x-pack/legacy/plugins/canvas/canvas_plugin_src/renderers/time_filter/components/datetime_calendar/datetime_calendar.tsx

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { FunctionComponent } from 'react';
+import PropTypes from 'prop-types';
+import { Moment } from 'moment';
+import { momentObj } from 'react-moment-proptypes';
+import { EuiDatePicker } from '@elastic/eui';
+import { DatetimeInput } from '../datetime_input';
+
+export interface Props {
+  /** Selected date (Moment date object) */
+  value?: Moment;
+  /** Function invoked when a date is selected from the datepicker */
+  onSelect: (date: Moment | null) => void;
+  /** Function invoked when the date text input changes */
+  onValueChange: (moment: Moment) => void; // Called with a moment
+  /** Start date of selected date range (Moment date object) */
+  startDate?: Moment;
+  /** End date of selected date range (Moment date object) */
+  endDate?: Moment;
+  /** Earliest selectable date (Moment date object) */
+  minDate?: Moment;
+  /** Latest selectable date (Moment date object) */
+  maxDate?: Moment;
+}
+
+export const DatetimeCalendar: FunctionComponent<Props> = ({
+  value,
+  onValueChange,
+  onSelect,
+  startDate,
+  endDate,
+  minDate,
+  maxDate,
+}) => (
+  <div className="canvasDateTimeCal">
+    <DatetimeInput moment={value} setMoment={onValueChange} />
+    <EuiDatePicker
+      inline
+      showTimeSelect
+      shadow={false}
+      selected={value && value.isValid() ? value : null}
+      onChange={onSelect}
+      shouldCloseOnSelect={false}
+      startDate={startDate}
+      endDate={endDate}
+      minDate={minDate}
+      maxDate={maxDate}
+    />
+  </div>
+);
+
+DatetimeCalendar.propTypes = {
+  value: PropTypes.oneOfType([momentObj, PropTypes.object]), // Handle both valid and invalid moment objects
```

Comment:
@clintandrewhall for moment proptypes, I went the `react-moment-proptypes` approach (Mostly because with typescript, `Moment` is the typescript type and not the moment class so the `instanceOf(Moment)` approach won't work unless I'm missing something). The downside is that `react-moment-proptypes` checks not only if the prop is a moment but it [also checks if it's a valid moment])(https://github.com/CalebMorris/react-moment-proptypes/blob/master/src/moment-validation-wrapper.js#L11) which messes with the DatetimeCalendar component since it's supposed to be able to handle invalid moments.

My compromise was to make the `value` proptype be either a moment or an object but I don't love it. I could make a custom validator proptype for value but it'd be only for just this single prop. Any thoughts or suggestions?